client/js/editmode.js

```diff
@@ -251,7 +259,7 @@ function populateAddWidgetOverlay() {
   });
 
   addCompositeWidgetToAddWidgetOverlay(generateCardDeckWidgets('add-deck', x, 500), function() {
-    for(const w of generateCardDeckWidgets(Math.random().toString(36).substring(3, 7), x, 500))
+    for(const w of generateCardDeckWidgets(generateUniqueWidgetID(), x, 500))
```

Comment:
This one and the counter use the ID for several widgets by appending fixed strings. So this isn't 100% guaranteed.

Reply:
Agreed. I did think about that, but the only way it could fail would be if a user invented a random four-character string, appended one of the magic characters like D, and gave a widget that id. Didn't seem worth addressing.